packages/background/src/controllers/NetworkController.ts

```diff
@@ -164,6 +165,36 @@ export default class NetworkController extends BaseController<NetworkControllerS
         return `CHAIN-${chainId}`;
     }
 
+    /**
+     * Change list of networks order.
+     */
+    public editNetworksOrder(newNetworksOrder: EditNetworksOrder): void {
+        const newNetworks = { ...this.networks };
+        newNetworksOrder.networksOrder.forEach((networkOrderUpdate) => {
+            const { chainId, order } = networkOrderUpdate;
+
+            //Validations
+            if (!chainId || Number.isNaN(chainId)) {
+                throw new Error('ChainId is required and must be numeric.');
+            }
+
+            const existingNetwork = this.getNetworkFromChainId(chainId);
+
+            if (!existingNetwork) {
+                throw new Error(
+                    'The network you are trying to edit does not exist.'
+                );
+            }
+
+            const networkKey = this._getNetworkKey(existingNetwork);
+
+            newNetworks[networkKey].order = order;
+        });
+
+        this.networks = newNetworks;
```

Comment:
This two lines updates the networks state twice. Couldn't that be done once? 

Maybe change the `orderNetworks` to something like `getSortedNetworks(networks:Networks) : Networks` or `normalizeNetowrksOrder(networks:Networks) : Networks `  or something similar?

and then you can update just once `this.networks = normalizeNetowrksOrder(newNetworks);`

Reply:
Great Idea! Done that and added it to utils/networks.ts.